### PR TITLE
Added frame functionality

### PR DESCRIPTION
**Pull request by trym-b**
added tests for:
    getting point cloud as buffer from frame
    releasing frame
    init of frame given path/str
    save frame to file
    using context manager to release frame

Deletes sample file after testing

### CODE REVIEW

File: test/test_frame.py [hunk 1]
```diff
@@ -2,11 +2,68 @@
...
+def test_release(frame):
+    frame.point_cloud()
+    frame.release()
+    with pytest.raises(RuntimeError):
+        frame.point_cloud()
+
+
+def test_path_init(application, sample_data_file):  # pylint: disable=unused-argument
+    import zivid
+
```
Comment (nedrebo) [inline test/test_frame.py]: Maybe assert that sample_data_file instance is Path?
Else this can easily be same test as the one below.

File: zivid/frame.py [hunk 2]
```diff
@@ -1,14 +1,30 @@
...
 class Frame:  # pylint: disable=too-few-public-methods
     def __init__(self, internal_frame):
-        if not isinstance(internal_frame, _zivid.Frame):
+        if isinstance(internal_frame, _zivid.Frame):
+            self.__frame = internal_frame
+        elif isinstance(internal_frame, (str, Path)):
+            self.__frame = _zivid.Frame(str(internal_frame))
+        else:
             raise ZividError(
                 f"Unsupported type for argument internal frame: {internal_frame}, type: {type(internal_frame)}."
```
Comment (nedrebo) [inline zivid/frame.py]: I would write internal_frame instead of internal frame.
I would also swap format strings since value may be large, then readability is better if it is at the end. Also I quote the values when serializing them is they can be empty. If you know it is an it you do not need to for example, but I guess i python you generally do not know, at least strings can be empty.

`f"Unsupported type for argument internal_frame: type: {type(internal_frame)}, value: \"{internal_frame}\"."` 

I also some times use {} instead of quotes. then I guess the format string would be `{{{internal_frame}}}`. We should standardize on things like this for this project. I don't have strong preferences for wither, but consistency is important and error messages with "missing text" is annoying.

In this case I actually think you do not need to include the value since it is not the problem, only the type is the reason for the exception.

File: zivid/frame.py [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
             )
-        self.__frame = internal_frame
+
+    def point_cloud(self):
+        return self.__frame.get_point_cloud()
```
Comment (nedrebo) [inline zivid/frame.py]: Here we are leaking _zivid. May be hard to fix, but we should try. Add a Todo:

File: zivid/frame.py (hunk 2 above)
Comment (trym-b) [reply zivid/frame.py]: How about creating an ZividUnsupportedTypeError?
or are you completely against different error types in python?
Could be something like this:
raise ZividUnsupportedTypeError(argument, expected_type)

File: test/test_frame.py (hunk 1 above)
Comment (trym-b) [reply test/test_frame.py]: added assert

File: zivid/frame.py (hunk 3 above)
Comment (trym-b) [reply zivid/frame.py]: Added a test that is marked with xfail

File: test/conftest.py [hunk 4]
```diff
@@ -16,9 +16,20 @@ def sample_data_file():
...
 @pytest.fixture()
 def file_camera(application, sample_data_file):  # pylint: disable=redefined-outer-name
     with application.create_file_camera(sample_data_file) as file_cam:
         yield file_cam
+
+
+@pytest.fixture()
+def frame(
+    application, sample_data_file
+):  # pylint: disable=redefined-outer-name, unused-argument
```
Comment (nedrebo) [inline test/conftest.py]: Move up

File: test/test_frame.py [hunk 5]
```diff
@@ -2,11 +2,83 @@
 
 
 def test_illegal_init(application):  # pylint: disable=unused-argument
+    import _zivid
     import zivid
 
-    with pytest.raises(zivid.errors.ZividError):
+    with pytest.raises(_zivid.ZividError):
```
Comment (nedrebo) [inline test/test_frame.py]: Why did you change one of these?

File: test/test_frame.py [hunk 6]
```diff
@@ -2,11 +2,83 @@
...
     with pytest.raises(zivid.errors.ZividError):
         zivid.frame.Frame(None)
     with pytest.raises(zivid.errors.ZividError):
         zivid.frame.Frame(12345)
+
+
+def test_point_cloud(frame):
+    import numpy as np
+
+    point_cloud = frame.point_cloud()
```
Comment (nedrebo) [inline test/test_frame.py]: Missing isinstance(... PointCloid)

the two following lines should probably be in test_point_cloud.py:
    np_array = np.array(point_cloud)
    assert isinstance(np_array, np.ndarray)

File: test/test_frame.py [hunk 7]
```diff
@@ -2,11 +2,83 @@
...
+    assert frame is not None
+    assert isinstance(frame, zivid.frame.Frame)
+
+
+def test_save(frame):
+    import tempfile
+    from pathlib import Path
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        save_path = Path(temp_dir) / "save_test.zdf"
```
Comment (nedrebo) [inline test/test_frame.py]: Add assert not save_path.exists()

File: test/test_frame.py [hunk 8]
```diff
@@ -2,11 +2,83 @@
...
+        frame.save(save_path)
+        assert save_path.exists()
+
+
+def test_context_manager(
+    application, sample_data_file
+):  # pylint: disable=unused-argument
```
Comment (nedrebo) [inline test/test_frame.py]: Up

File: zivid/__init__.py [hunk 9]
```diff
@@ -7,3 +7,4 @@
 from zivid.settings import (  # noqa: F401 'zivid.settings.Settings' imported but unused
     Settings,
 )
+from zivid.frame import Frame  # noqa: F401 'zivid.frame.Frame' imported but unused
```
Comment (nedrebo) [inline zivid/__init__.py]: Why (..) above and not here?

File: test/test_frame.py (hunk 5 above)
Comment (trym-b) [reply test/test_frame.py]: Frame can now be constructed from path, so if you pass in an invalid path you get _zividerror instead

File: test/test_frame.py (hunk 7 above)
Comment (trym-b) [reply test/test_frame.py]: You don't trust temporary directory?

File: zivid/__init__.py (hunk 9 above)
Comment (trym-b) [reply zivid/__init__.py]: I don't get what you are saying here. What (..)?

File: test/test_frame.py (hunk 6 above)
Comment (trym-b) [reply test/test_frame.py]: fixed, added test_point_cloud.py
